packages/@aws-cdk/assets/lib/asset.ts

```diff
@@ -221,73 +221,18 @@ export class Asset extends cdk.Construct implements IAsset {
   }
 }
 
-export interface FileAssetProps {
-  /**
-   * File path.
-   */
-  readonly path: string;
-
-  /**
-   * A list of principals that should be able to read this file asset from S3.
-   * You can use `asset.grantRead(principal)` to grant read permissions later.
-   *
-   * @default - No principals that can read file asset.
-   */
-  readonly readers?: iam.IGrantable[];
-}
-
-/**
- * An asset that represents a file on disk.
- */
-export class FileAsset extends Asset {
-  constructor(scope: cdk.Construct, id: string, props: FileAssetProps) {
-    super(scope, id, { packaging: AssetPackaging.File, ...props });
+function determinePackaging(assetPath: string): AssetPackaging {
+  if (!fs.existsSync(assetPath)) {
+    throw new Error(`Cannot find asset at ${assetPath}`);
   }
-}
-
-export interface ZipDirectoryAssetProps {
-  /**
-   * Path of the directory.
-   */
-  readonly path: string;
 
-  /**
-   * A list of principals that should be able to read this ZIP file from S3.
-   * You can use `asset.grantRead(principal)` to grant read permissions later.
-   *
-   * @default - No principals that can read file asset.
-   */
-  readonly readers?: iam.IGrantable[];
-}
-
-/**
- * An asset that represents a ZIP archive of a directory on disk.
- */
-export class ZipDirectoryAsset extends Asset {
-  constructor(scope: cdk.Construct, id: string, props: ZipDirectoryAssetProps) {
-    super(scope, id, { packaging: AssetPackaging.ZipDirectory, ...props });
+  if (fs.statSync(assetPath).isDirectory()) {
```

Comment:
Would like for this to feature & adhere to symlink control behavior...

Reply:
Not sure what you mean. `stat` follows symlinks, so if the path is a symlink to a directory or file, it should just work.

Reply:
What I mean is I might want to prohibit symlinking to `/etc/passwd` for example... Or taking files from `/lib64` (which I have encountered before and can cause a mountain of problems).

Basically, I want to control Symlink traversal here... so that I can choose where files may come from. And if I say "no symlink out of app root", then if the asset lives out of there you should fail on the spot.